### PR TITLE
added a function add_n

### PR DESCRIPTION
**Pull request by princebari**
close https://github.com/unifyai/ivy/issues/16748

### CODE REVIEW

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi! Thanks a lot for looking into this 😄. Not sure which task board this belongs to, could you please add the appropriate comments so that your PR gets correctly linked? :)
Feel free to refer to the docs on the comment format - https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues

**Comment by princebari** [comment]
close ##16748

**Comment by princebari** [comment]
close #16748

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def log10(self, name=None):
     def argsort(self, axis=-1, descending=False, name=None):
         return ivy.argsort(self._ivy_array, axis=axis, descending=descending)
 
-    @with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
+    @with_unsupported_dtypes({"2.4.2 and below": ("float16", "float16")}, "paddle")
```
Comment (rishabgit) [inline ivy/functional/frontends/paddle/tensor/tensor.py:149]: this needs to be reverted

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 2]
```diff
@@ -158,6 +158,10 @@ def tanh(self, name=None):
     def add_(self, name=None):
         return ivy.add(self._ivy_array)
 
+    @with_supported_dtypes({"2.4.3 and below": ("float32", "float64")}, "paddle")
+    def add_n(self, name= None):
+        return ivy.add_n(self._ivy_array)
```
Comment (rishabgit) [inline ivy/functional/frontends/paddle/tensor/tensor.py:163]: ivy functional api doesn't have add_n, supported dtypes need update too

**Review by rishabgit — CHANGES_REQUESTED** [review]
Hi! Your code seems to call ivy.add_n which doesn't exist 😅 I'd heavily suggest going through these deep dive resources to help you out - https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#ivy-frontends, https://unify.ai/docs/ivy/overview/deep_dive/function_types.html
Also, I looked into Paddle's docs - https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/add_n_en.html#add-n and the supported dtypes needs an update too
Feel free to re-request review after making these changes, thanks! 😄

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by princebari** [comment]
Dear Rishab

I am writing to bring to your attention an issue regarding the implementation of the "add_n" function in the Ivy backend. After attempting to utilize this function, I have discovered that it is currently not available or supported.

I would greatly appreciate it if you could look into this matter and consider implementing the "add_n" function. If there are any updates or plans for future additions to the backend, I would be grateful to receive any information regarding the availability of this function.

Thank you for your attention to this matter. I look forward to hearing from you soon and resolving this issue.

Best regards,
Prince

**Comment by rishabgit** [comment]
> Dear Rishab
> 
> I am writing to bring to your attention an issue regarding the implementation of the "add_n" function in the Ivy backend. After attempting to utilize this function, I have discovered that it is currently not available or supported.
> 
> I would greatly appreciate it if you could look into this matter and consider implementing the "add_n" function. If there are any updates or plans for future additions to the backend, I would be grateful to receive any information regarding the availability of this function.
> 
> Thank you for your attention to this matter. I look forward to hearing from you soon and resolving this issue.
> 
> Best regards, Prince

Hi, not sure if this is an issue of adding `add_n` in Ivy framework. Apart from the resources I shared above, could you please check out https://unify.ai/docs/ivy/overview/deep_dive/superset_behaviour.html to decide what approach makes the most sense for this task? Thanks!

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**